packages/web/src/components/layout/AppLayout/NavBar/NavDrawer.tsx

```diff
@@ -1,91 +1,59 @@
 import { useRouter } from 'next/router';
-import { signInWithSlack } from './NavBar';
+import { MdClose } from 'react-icons/md';
 import { useUserStore } from '../../../../stores/user';
+import { Logout } from './NavElements/AuthButtons/Logout';
+import { NavLogo } from './NavLogo';
+import { Schedule } from './NavElements/PageLinks/Schedule';
+import { useAdminStore } from '../../../../stores/admin';
+import { AdminDashboard } from './NavElements/PageLinks/AdminDashboard';
 
 type NavDrawerProps = {
-  isOpen: any;
-  onClose: any;
+  isOpen: boolean;
+  onClose: () => void;
 };
 
 export const NavDrawer: React.FC<NavDrawerProps> = ({ onClose, isOpen }) => {
   const router = useRouter();
   const { user, doneLoading } = useUserStore();
+  const { admin } = useAdminStore();
+
+  React.useEffect(() => {
+    // Close the menu if the route changes
+    router.events.on('routeChangeStart', onClose);
+  }, [onClose, router.events]);
 
   return (
     <Drawer placement="left" size={'xs'} onClose={onClose} isOpen={isOpen}>
       <DrawerOverlay />
+
       <DrawerContent>
         <DrawerBody>
           <Fade in={doneLoading}>
-            <Flex direction="column" gap={5} alignItems="center">
-              {user?.firstName ? (
-                <Box alignContent={'center'}>Welcome back, {user.firstName}!</Box>
-              ) : (
-                <>
-                  <Button
-                    variant="cta"
-                    onClick={async () => {
-                      await signInWithSlack();
-                    }}
-                  >
-                    Sign Up
-                  </Button>
-                  <Button
-                    onClick={async () => {
-                      await signInWithSlack();
-                    }}
-                  >
-                    Login
-                  </Button>
-                </>
-              )}
-              <Link
-                onClick={() => {
-                  void router.push('/schedule');
-                  onClose();
-                }}
-              >
-                Schedule
-              </Link>
-              <Link
-                onClick={() => {
-                  void router.push('/admin/dashboard');
-                  onClose();
-                }}
-              >
-                Admin Dashboard
-              </Link>
+            <Flex direction="column" gap={5} alignItems="left">
+              <Flex position="relative" w="full" justify="center" alignItems="center">
+                <NavLogo />
+
+                <IconButton aria-label="Close nav button" icon={<MdClose />} onClick={onClose} />
+              </Flex>
 
-              {user && (
-                <Button
-                  onClick={() => {
-                    window.location.href = '/api/auth/logout';
-                  }}
-                >
-                  {`Logout`}
-                </Button>
-              )}
+              <Schedule />
+
+              {admin && <AdminDashboard />}
```

Comment:
This just says "If admin exists, show <AdminDashboard /> right?"

Reply:
Yessir!